db/migrate/20160527140046_foreign_keys.rb

```diff
@@ -0,0 +1,116 @@
+class ForeignKeys < ActiveRecord::Migration
+  class ClassificationSubject < ActiveRecord::Base
+  end
+
+  def up
+    add_foreign_key :access_control_lists, :user_groups, on_update: :cascade, on_delete: :cascade
+
+    Aggregation.joins("LEFT OUTER JOIN workflows ON workflows.id = aggregations.workflow_id").where("workflows.id IS NULL").delete_all
+    add_foreign_key :aggregations, :workflows, on_update: :cascade, on_delete: :cascade
+    add_foreign_key :aggregations, :subjects, on_update: :cascade, on_delete: :cascade
+
+    add_foreign_key :authorizations, :users, on_update: :cascade, on_delete: :cascade
+
+    # Classification foreign keys are done all the way at the end because we might be deleting projects or workflows
+    # that they refer to on the following lines. Apart from that, everything is in alphabetical order, same as the structure.sql
+
+    # Commented out add_foreign_key calls are ones that already exist. Left them in here so that it's
+    # easier to check if I caught all of them.
+    #
+    # add_foreign_key :classification_subjects, :classifications
+    # add_foreign_key :classification_subjects, :subjects
+
+    add_foreign_key :collections_subjects, :subjects, on_update: :cascade, on_delete: :restrict
+    add_foreign_key :collections_subjects, :collections, on_update: :cascade, on_delete: :restrict
```

Comment:
Ah, I see now. Yeah, we should `cascade` the deletion of a collection rather than restrict.